Makefile

```diff
@@ -347,6 +348,22 @@ e2e: # Run E2E integration tests.
 	./hack/verify.sh integration
 .PHONY: e2e
 
+run-on-kind: ## Run Prometheus, Grafana and Kepler on Kind, quick dev environment
+	CLUSTER_PROVIDER=$(CLUSTER_PROVIDER) \
+	LOCAL_DEV_CLUSTER_VERSION=$(LOCAL_DEV_CLUSTER_VERSION) \
+	KIND_WORKER_NODES=$(KIND_WORKER_NODES) \
+	BUILD_CONTAINERIZED=$(BUILD_CONTAINERIZED) \
+	PROMETHEUS_ENABLE=true \
+	GRAFANA_ENABLE=true \
+	./hack/cluster.sh up
+	OPTS=$(DEV_OPTS) \
+	BUILD_CONTAINERIZED=$(BUILD_CONTAINERIZED) \
+	./hack/cluster-deploy.sh
```

Comment:
make [cluster-deploy](https://github.com/sustainable-computing-io/kepler/blob/main/hack/cluster-deploy.sh) extactly does the same thing:  builds and deploy kepler on kind 

`make cluster-up cluster-deploy` will do the job

Reply:
`make cluster-up cluster-deploy` Doesn't automatically deploy Prometheus + Grafana + Kind into the same cluster... it also builds 2 container images not just a basic developer one without DCGM.

 I'm just trying to create an easier point of access for people to get started with the project -  if this is not desired I'm happy to drop this change.

Reply:
> make cluster-up cluster-deploy Doesn't automatically deploy Prometheus + Grafana + Kepler into the kind cluster... it also builds 2 container images not just a basic developer one without DCGM.

A quick way to address this can be:

Fix in cluster.sh: 
```
declare -r GRAFANA_ENABLE=${GRAFANA_ENABLE:-false}
....
....
export GRAFANA_ENABLE
```
In cluster-up target set GRAFANA_ENABLE as `true` ensuring both Prometheus and Grafana get installed:
```
GRAFANA_ENABLE=true
...
GRAFANA_ENABLE=$(GRAFANA_ENABLE) \
KIND_WORKER_NODES=$(KIND_WORKER_NODES) \
./hack/cluster.sh up
```
For only building Kepler(without dcgm) small update in cluster-deploy.sh
```
declare KEPLER_BUILD_DCGM="${KEPLER_BUILD:-false}"
...
local kepler_build_target="build_image"
$KEPLER_BUILD_DCGM && {
	kepler_build_target="build_image_dcgm"
}
run make "$kepler_build_target" \
	IMAGE_REPO="$IMAGE_REPO" \
	IMAGE_TAG="$IMAGE_TAG" \
	VERSION="devel"
```
by default, it will build Kepler without dcgm and if someone wants to play with dcgm they can pass the value during runtime `make cluster_deploy KEPLER_BUILD_DCGM=true`
@maryamtahhan Wdyt?

Reply:
I did think of that option but had opted against it as targets in the Makefile are already in use with a specific intention. If we think we are OK with changing the existing intention then I've no issue making the changes.

There is an option to use [Target-specific Variable Values](https://www.gnu.org/software/make/manual/html_node/Target_002dspecific.html) in the Makefile. 

So we would re-use the cluster-up target under run-on-kind

```
run-on-kind: export PROMETHEUS_ENABLE=true
run-on-kind: export GRAFANA_ENABLE=true
run-on-kind: cluster-up ## Run Prometheus, Grafana and Kepler on Kind, quick dev environment
	OPTS=$(DEV_OPTS) \
	BUILD_CONTAINERIZED=$(BUILD_CONTAINERIZED) \
	./hack/cluster-deploy.sh
.PHONY: run-on-kind
```
we wouldn't use cluster-deploy directly as this can deploy into a physical cluster as well as kind which is why I invoke it directly under the run-on-kind target

Reply:
The [above](https://github.com/sustainable-computing-io/kepler/pull/1337#discussion_r1575176743) changes don't impact any existing functionality. It only allows Grafana to be deployed in Kind cluster and makes the image build step variable in `cluster-deploy.sh` Instead of creating a separate make target I think it's best to enhance the existing ones whenever possible. 🙂

Reply:
> The [above](https://github.com/sustainable-computing-io/kepler/pull/1337#discussion_r1575176743) changes don't impact any existing functionality. It only allows Grafana to be deployed in Kind cluster and makes the image build step variable in `cluster-deploy.sh` Instead of creating a separate make target I think it's best to enhance the existing ones whenever possible

I disagree the cluster up target is already in use with a specific intent, with the suggested change the intent is different and has a usage impact, with the script changes:

| old intent | new intent | impact |
| -------- | ------- |------- |
| Create a kind cluster | Create a kind cluster and deploy grafana + prometheus | different (more) resource usage wherever this is already used including CI pipelines |

Re the cluster-deploy.sh - making the image build step variable is fine but this alone doesn't solve the other aspect which is needing the OPTS="PROMETHEUS_DEPLOY" to do a Prometheus/grafana manifest build under the cluster-deploy target. From earlier on in the Makefile in this PR we are setting the additional `DEV_OPTS ?= "CI_DEPLOY PROMETHEUS_DEPLOY"`

the idea with the `run-on-kind` target is to create a quick dev env/validation env where everything is deployed inside kind... with the suggestion using the target specific variables we are re-using the older targets. 

I agree with @SamYuan1990 that a discussion around the targets and the scripts should be revisited as these seem to be peppered in scripts and the Makefiles. 

My feedback here is there is not a simple getting started point for onboarding new developers. The first thing any new developer is going to do is try to deploy the whole stack (Kepler + Prometheus + Grafana) on kind to get a feel for the project. Right now they need to read 2 different parts of the documentation and understand what env vars and script vars to set to get that stack up and running (that's fine - but it's slow and would probably involve some trial and error to get the exact commands right). I think we are all in agreement on this. But I would argue that whatever changes are made to scripts/deployment intent need backwards compatibility as they can impact existing users and their pipelines.

For now I will try to update the documentation with a one liner for new developers...

Reply:
@maryamtahhan Highly Appreciate the feedback 🙂  .  I completely agree with the documentation part. It's very confusing, outdated and needs updating. 

> I disagree the cluster up target is already in use with a specific intent, with the suggested change the intent is different and has a usage impact, with the script changes:

AFAIK we only use `make cluster-up` while setting up things locally or in `.github/workflows/developer_local.yml`. I don't see enabling Grafana/Prometheus would impact resources, especially on CI. I completely agree with the notion of having a make target that deploys a complete infra stack (Kind + Prometheus + Grafana). The intent of `make cluster-up` is to do so. That's why I suggested to amend the existing target so that it creates less confusion. Having a target that only deploys the Kind cluster and another that deploys `Kind + monitoring stack` creates more friction. Ideally monitoring stack should be enabled from the beginning.